base_layer/core/src/chain_storage/blockchain_database.rs

```diff
@@ -406,6 +411,34 @@ where T: BlockchainBackend
         fetch_utxo(&*db, hash)
     }
 
+    /// Returns all UTXOs
+    pub fn fetch_all_utxos(&self) -> Result<Vec<TransactionOutput>, ChainStorageError> {
+        let db = self.db_read_access()?;
+        let utxo_count = db.count_utxos()?;
+        let mut outputs = Vec::with_capacity(utxo_count);
+        db.for_each_utxo(|utxo| {
+            if let Ok((_, output)) = utxo {
+                outputs.push(output);
+            }
+        })?;
+
+        Ok(outputs)
+    }
+
+    /// Returns all kernels
+    pub fn fetch_all_kernels(&self) -> Result<Vec<TransactionKernel>, ChainStorageError> {
+        let db = self.db_read_access()?;
+        let kernel_count = db.count_kernels()?;
+        let mut kernels = Vec::with_capacity(kernel_count);
+        db.for_each_kernel(|kernel| {
+            if let Ok((_, kernel)) = kernel {
+                kernels.push(kernel);
+            }
+        })?;
+
+        Ok(kernels)
+    }
```

Comment:
Are we sure we want to add these calls? They can be massive? 
Effectively we ask for the entire database in a single call

Reply:
You're right we need an iterator - this was a stop gap to implementing that

Reply:
I think I have a good "chunked" (bounded memory) iterator implementation for LMDB, just have to write tests to see if it works. My feeling is that this part can stay as is for this PR and optimised soon after. This implementation doesn't break on testnet when I tested it, though it will cause a (short?) spike in memory while the kernels/UTXO commitments are summed.

Reply:
We should not implement a iterator over the db. We dont know the db backend, and if we can implement a iterator. Just on top of my head. Does postgres have one?

Reply:
The idea is to load in chunks, which every database should be able to do.